File: app/src/main/java/com/designpatterns/demo/factory/java/FactoryA.java
package com.designpatterns.demo.factory.java;

/**
 * 建具体工厂类（继承抽象工厂类），定义创建对应具体产品实例的方法
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/5/17 下午2:14
 * @project_name DesignPatternsDemo
 */
public class FactoryA extends Factory {
    @Override
    public Product Manufacture() {
        return new ProductA();
    }
}

File: app/src/main/java/com/designpatterns/demo/factory/kotlin/FactoryA.kt
package com.designpatterns.demo.factory.kotlin

import com.designpatterns.demo.factory.java.Factory
import com.designpatterns.demo.factory.java.Product
import com.designpatterns.demo.factory.java.ProductA

/**
 * 创建具体工厂类（继承抽象工厂类），定义创建对应具体产品实例的方法
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/5/17 下午3:11
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */
class FactoryA : Factory() {
    override fun Manufacture(): Product = ProductA()
}